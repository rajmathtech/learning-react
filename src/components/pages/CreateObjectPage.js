import React from 'react';
import {connect} from 'react-redux';
import ObjectForm from '../ObjectForm';
import { startAddObject } from '../../redux/actions/object';
export class CreateObjectPage extends React.Component{
    onSubmitForm= (object) => {
        this.props.startAddObject(object);
        this.props.history.push('/'); 
     };
    render () {
        return (<div className="container__main">
        <h1 className="heading-primary"> Welcome to Create Object</h1>
        <ObjectForm  object={{}} onSubmitForm={this.onSubmitForm}/>
        </div>)
    }
} 

const mapDispatchToProps = (dispatch) => ({
    startAddObject:(object)=>dispatch(startAddObject(object))
});

export default connect(undefined, mapDispatchToProps)(CreateObjectPage);