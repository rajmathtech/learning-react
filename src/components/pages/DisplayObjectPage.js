import React from 'react';
import { Link } from 'react-router-dom';
import { startDeleteObject } from '../../redux/actions/object';
import { connect } from 'react-redux';
import moment from 'moment';
import numeral from 'numeral';
const DisplayObjectPage = (props) => {
    const onDeleteObject = () => {
        props.dispatch(startDeleteObject(props.object.id));  
        props.history.push('/');  
    }
    const {id, title, details, amount, createdAt} = props.object;
    return  (<div className="popup"> 
        <div className="popup__content">
            <h1> Welcome to Edit Object Page</h1>
            <h2>{title}</h2>
            <h2>{details}</h2>
            <h3>{moment(createdAt).format('MMMM Do, YYYY')}</h3>
            <h3>{numeral(amount/100).format('$0,0.00')}</h3>
            <Link to="/" className="editLink"> Go back</Link>
            <Link to={`/edit/${id}`} className="editLink"> Edit</Link>
            <button onClick={onDeleteObject}> Delete </button>
        </div>  
    </div>);
};
    
const mapStateToProps = (state, props)=>
{ 
    return {
        object:state.objects.find((obj) => 
        {
            if (obj.id === props.match.params.id) {
                return true;
            }
        })
    };
};
export default connect(mapStateToProps)(DisplayObjectPage);
