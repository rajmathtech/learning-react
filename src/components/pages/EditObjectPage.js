import React from 'react';
import ObjectForm from '../ObjectForm';
import {connect} from 'react-redux';
import { editObject }  from '../../redux/actions/object';
export class EditObjectPage extends React.Component{
    onSubmitForm = (update) => {    
        this.props.editObject(this.props.object, update);
        this.props.history.push('/');
    };
    render () {
        
    return (<div className="container__main">
    <h1 className="heading-primary"> Welcome to Edit Object</h1>
    <ObjectForm object={this.props.object} onSubmitForm={this.onSubmitForm}/>
    </div>);
    }
}
const mapDispatchToProps = (dispatch) => ({
    editObject: (obj, object) => dispatch(editObject(obj, object))
})
 const mapStateToProps = (state, props) =>({
    object:state.objects.find((object) => object.id === props.match.params.id)
 });

export default connect(mapStateToProps, mapDispatchToProps)(EditObjectPage);