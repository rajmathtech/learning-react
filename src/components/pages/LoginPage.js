import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../../redux/actions/auth';
export const LoginPage = ({startLogin}) => (
    <div> 
    <h1> welcome to Log in Page </h1>
    <button onClick={startLogin}> LogIn </button>
    </div>
);
const mapDispatchToProps = (dispatch) =>{
    return {
        startLogin: () => dispatch(startLogin())
    }
}
export default connect(undefined, mapDispatchToProps)(LoginPage);