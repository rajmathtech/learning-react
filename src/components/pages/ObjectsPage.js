import React from 'react';
import Objects from '../Objects';
const ObjectsPage = () => (<div className="container__main"> 
    <h1 className="heading-primary"> Welcome to Objects list </h1> 
    <Objects />
</div>);

export default ObjectsPage;