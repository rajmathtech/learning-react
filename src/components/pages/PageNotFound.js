import React from 'react';

const PageNotFound = () => (<div> <h2> 404- page not found! </h2> </div>);

export default PageNotFound;