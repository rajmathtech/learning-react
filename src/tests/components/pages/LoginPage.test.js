import React from 'react';
import {shallow} from 'enzyme';
import {LoginPage} from '../../../components/pages/LoginPage';
test('should render Login page component', () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});
test('should call startLogin spy', () => {
    const startLogin = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={startLogin}/>);
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
});