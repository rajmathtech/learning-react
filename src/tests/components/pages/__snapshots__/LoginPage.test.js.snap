// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login page component 1`] = `
<div>
  <h1>
     welcome to Log in Page 
  </h1>
  <button>
     LogIn 
  </button>
</div>
`;
